subgraphs/compound-v3/src/sdk/account.ts

```diff
@@ -0,0 +1,331 @@
+import {
+  BigDecimal,
+  BigInt,
+  Bytes,
+  ethereum,
+  log,
+} from "@graphprotocol/graph-ts";
+import {
+  Account,
+  LendingProtocol,
+  Market,
+  Position,
+  PositionSnapshot,
+  PositionCounter,
+} from "../../generated/schema";
+import {
+  BIGINT_ZERO,
+  exponentToBigDecimal,
+  INT_ONE,
+  INT_ZERO,
+  TransactionType,
+} from "./constants";
+import { TokenManager } from "./token";
+
+/**
+ * This file contains the AccountClass, which does
+ * the operations on the Account entity. This includes:
+ *  - Creating a new Account
+ *  - Updating an existing Account
+ *  - Making a position
+ *  - Making position snapshots
+ *
+ * Schema Version: 3.0.0
+ * Last Updated: Dec 4, 2022
+ * Author(s):
+ *  - @dmelotik
+ */
+export class AccountManager {
+  private isNew: boolean; // true if the account was created
+  private account: Account;
+  private positonError: boolean;
+  private position!: Position;
+  private market: Market;
+  private protocol: LendingProtocol;
+  private event: ethereum.Event;
+
+  constructor(
+    account: Bytes,
+    market: Market,
+    protocol: LendingProtocol,
+    event: ethereum.Event
+  ) {
+    this.market = market;
+    this.protocol = protocol;
+    this.event = event;
+    let _account = Account.load(account);
+    if (!_account) {
+      _account = new Account(account);
+      _account.positionCount = INT_ZERO;
+      _account.openPositionCount = INT_ZERO;
+      _account.closedPositionCount = INT_ZERO;
+      _account.depositCount = INT_ZERO;
+      _account.withdrawCount = INT_ZERO;
+      _account.borrowCount = INT_ZERO;
+      _account.repayCount = INT_ZERO;
+      _account.liquidateCount = INT_ZERO;
+      _account.liquidationCount = INT_ZERO;
+      _account.transferredCount = INT_ZERO;
+      _account.receivedCount = INT_ZERO;
+      _account.flashloanCount = INT_ZERO;
+      _account.save();
+      this.isNew = true;
+    } else {
+      this.isNew = false;
+    }
+    this.account = _account;
+  }
+
+  // returns true if the account was created in this instance
+  isNewAccount(): boolean {
```

Comment:
Since you already have the protocol entity when initializing the account, you can update the unique users counters directly from here instead of exposing an `isNewAccount` method. Let the caller forget about those metrics

Reply:
I guess this is the same thing as this: https://github.com/messari/subgraphs/pull/1357#discussion_r1041654183

Reply:
Yeah, my bad. Resolving this one